### DISCUSSION TITLE
Can a programmer work without SD card?

### DISCUSSION THREAD

**Discussion by Paar86** [sanni/cartreader#105] [Need help]
Hi. I have built a specialized programmer based on the Card Reader project that should work only with 29F1610 chips. Below you can see how it looks like. I have successfully fleshed my Arduino with Card Reader code while commenting out the enable_OLED variable and currently am getting SD card error message in serial monitor. Is it possible to use this programmer without SD card reader and flash chips with files directly from my PC?

![20210305_150556](https://user-images.githubusercontent.com/34010694/110126806-19370e80-7dc5-11eb-8e8e-b1c8ecdde9b8.jpg)

**Comment by sanni** [comment]
Reading files from a PC has not been implemented.

**Reply by Paar86** [reply]
Good to know, so I'll buy a card slot. Hopefully it will work without problem then. Thank you!

**Reply by prlaba** [reply]
Did you ever get this working? I'm looking to do something similar to program the 29F1610 Flash ROMs.

How did you generate the needed +10V for writing?

**Comment by Paar86** [comment]
Hi. The programmer is probably working fine, at least I can dump the same
data from the programmed EEPROM as from the original MASK ROM. I think you
don't need +10V for writing, in the spec sheets there is information
about +11-12V when getting manufacturer/chip ID.

st 20. 10. 2021 v 4:13 odesílatel prl519 ***@***.***> napsal:

**Comment by sanni** [comment]
Die 12V is only needed for reading the ID when you don't know what flashrom you are dealing with. It's basically an universal ID command.
I'm using the second way of getting the ID by using the specific flashrom commands listed in the datasheet which doesn't need the 12V.
